polus/plugins.py

```diff
@@ -720,6 +724,123 @@ def update_nist_plugins(gh_auth: typing.Optional[str] = None):
             _error_log(val_err, manifest)
 
 
+class Registry:
+    """Class that contains methods to interact with the REST API of WIPP Registry."""
+
+    def get_current_schema(
+        self,
+        registry_url="https://wipp-registry.ci.aws.labshare.org/",
+        verify: bool = True,
+    ):
+        """Return current schema in WIPP"""
+        response = requests.get(
+            urljoin(
+                registry_url, "rest/template-version-manager/global/?title=res-md.xsd"
+            ),
+            verify=verify,
+        )
+        if response.ok:
+            return response.json()[0]["current"]
+        else:
+            response.raise_for_status()
+
+    def upload_data(
+        self,
+        filepath,
+        schema_id,
+        username,
+        password,
+        registry_url="https://wipp-registry.ci.aws.labshare.org/",
```

Comment:
Add `username`, `password`, and `registry_url` to an `__init__` method that stores them as class attributes. Then use them in all instance methods.